gladier_xpcs/flow_base.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+import random
+import globus_sdk
+from gladier import GladierBaseClient, utils
+# import gladier_xpcs.log  # Uncomment for debug logging
+
+log = logging.getLogger(__name__)
+
+
+class XPCSBaseClient(GladierBaseClient):
+    # Allow all admins of the XPCS developers group to deploy/run flows
+    globus_group = '368beb47-c9c5-11e9-b455-0efb3ba9a670'
+    # This is a bit of a hack while we wait for Globus SDK v3, after that
+    # it should automaically retry on these codes
+    TRANSIENT_ERROR_STATUS_CODES = (429, 500, 502, 503, 504)
+    max_retries = 10
+
+    def register_funcx_function(self, function):
+        """Register containers for any functions listed in self.containers"""
+        fxid_name = utils.name_generation.get_funcx_function_name(function)
+        if fxid_name in self.containers.keys():
+            container = self.containers[fxid_name]
+            log.info(f'Registering {fxid_name} with {container["location"]}')
+            fxid_name = utils.name_generation.get_funcx_function_name(function)
+            fxck_name = utils.name_generation.get_funcx_function_checksum_name(function)
+            cfg = self.get_cfg(private=True)
+            cid = self.funcx_client.register_container(**container)
+            fxid = self.funcx_client.register_function(function, function.__doc__, container_uuid=cid)
+            cfg[self.section][fxid_name] = fxid
+            cfg[self.section][fxck_name] = self.get_funcx_function_checksum(function)
+            cfg.save()
+        else:
+            super().register_funcx_function(function)
+
+    def retry_backoff(self, method, *args, **kwargs):
```

Comment:
I fell like this should go into gladier base.

Reply:
Really, it won't be needed in a week or so. I asked the automate flow about automatic retries, and it's automatic in the next version. We just need to wait a few days for them to do the stable release.